src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,115 @@
+    private static final String RIGHT_PASSWORD = "absabs";
+    private static final String LOGIN_DOES_NOT_EXIST = "abrakadabra";
+    private static final int WRONG_AGE = 17;
+    private static final String WRONG_PASSWORD = "12345";
+    private static RegistrationServiceImpl registration;
+    private User rightUserExist;
+    private User wrongUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registration = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        rightUserExist = new User();
+        rightUserExist.setLogin(RIGHT_LOGIN_EXIST);
+        rightUserExist.setAge(RIGHT_AGE);
+        rightUserExist.setPassword(RIGHT_PASSWORD);
+        new StorageDaoImpl().add(rightUserExist);
+        wrongUser = new User();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userExistTest() {
+        assertThrows(RuntimeException.class, () -> registration.register(rightUserExist));
+    }
+
+    @Test
+    void register_userIsRight_And_DoesNotExist() {
+        User rightDoesNotExist = new User();
+        rightDoesNotExist.setLogin(LOGIN_DOES_NOT_EXIST);
+        rightDoesNotExist.setAge(RIGHT_AGE);
+        rightDoesNotExist.setPassword(RIGHT_PASSWORD);
+        assertDoesNotThrow(() -> registration.register(rightDoesNotExist));
+    }
+
+    @Test
+    void validate_UserIsNull() {
+        assertThrows(RuntimeException.class, () -> registration.validatingUser(null));
+    }
+
+    @Test
+    void userLoginDoesNotExist() {
+        wrongUser.setLogin(LOGIN_DOES_NOT_EXIST);
+        assertDoesNotThrow(() -> registration.validatingUserLogin(wrongUser));
+    }
+
+    @Test
+    void userLoginExist() {
+        assertThrows(RuntimeException.class,
+                () -> registration.validatingUserLogin(rightUserExist));
+    }
+
+    @Test
+    void userLoginNull() {
+        wrongUser.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registration.validatingUserLogin(wrongUser));
+    }
+
+    @Test
+    void userAgeIsOk() {
+        assertDoesNotThrow(() -> registration.validatingUserAge(rightUserExist));
+    }
+
+    @Test
+    void userAgeLessThanMinimum() {
+        wrongUser.setAge(WRONG_AGE);
+        assertThrows(RuntimeException.class, () -> registration.validatingUserAge(wrongUser));
+    }
+
+    @Test
+    void userPasswordIsOK() {
+        assertDoesNotThrow(() -> registration.validatingUserPassword(rightUserExist));
+    }
+
+    @Test
+    void userPasswordLessThanMinimum() {
+        wrongUser.setPassword(WRONG_PASSWORD);
+        assertThrows(RuntimeException.class, () -> registration.validatingUserPassword(wrongUser));
+    }
+
+    @Test
+    void userPasswordIsNull() {
+        wrongUser.setPassword(null);
+        assertThrows(RuntimeException.class, () -> registration.validatingUserPassword(wrongUser));
+    }
+
+    @AfterAll
+    static void afterAll() {
+        Storage.people.clear();
+    }
```

Comment:
Aren't you repeating yourself? You have two methods for clearing the Storage - one is called after Each test, what means after the last test storage will be clear, and another one is clearing the storage ones, in the end, it's better to  remove one of them